src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +158,123 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
     }
 }
 
+void IInferRequestInternal::SetBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    OPENVINO_ASSERT(
+        !std::any_of(blobs.begin(),
+                     blobs.end(),
+                     [](const Blob::Ptr& item) {
+                         return !item || item->is<RemoteBlob>() || item->is<CompoundBlob>();
+                     }),
+        "set_input_tensors/set_tensors error. Default implementation doesn't support RemoteTensor or legacy "
+        "CompoundBlob objects");
+
+    OPENVINO_ASSERT(!blobs.empty(),
+                    "set_input_tensors/set_tensors can't be called with empty blobs for input '",
+                    name,
+                    "'");
+    if (blobs.size() == 1) {
+        SetBlob(name, blobs[0]);
+        return;
+    }
+    std::shared_ptr<const ov::op::v0::Parameter> param;
+    const auto& inputs = GetInputs();
+    for (const auto& input : inputs) {
+        if (auto p = std::dynamic_pointer_cast<const ov::op::v0::Parameter>(input)) {
+            const auto& names = p->output(0).get_tensor().get_names();
+            if (names.find(name) != names.end()) {
+                param = p;
+                break;
+            }
+        }
+    }
+    OPENVINO_ASSERT(param, "set_input_tensors/set_tensors error. Parameter '", name, "' is not found");
+    OPENVINO_ASSERT(ov::layout::has_batch(param->get_layout()),
+                    "set_input_tensors/set_tensors can be used only for inputs with N(batch) dimension"
+                    " 'layout' defined. Current layout for '",
+                    name,
+                    "' is ",
+                    param->get_layout().to_string());
```

Comment:
can we use tests for caching?
1. If we cache based on ov::Model with layout information, once we import_model we can propogate information about layouts from ov::Model to import_model as a WA
2. If we cache based on file_name, layouts information is lost after import_model and plugins need to fix it. Once this is implemented in plugins, we can remove WA in # 1 on Core level